ams/server/src/main/java/com/netease/arctic/server/optimizing/OptimizingQueue.java

```diff
@@ -308,19 +315,20 @@ SchedulingPolicy getSchedulingPolicy() {
 
   private class TableOptimizingProcess implements OptimizingProcess, TaskRuntime.TaskOwner {
     private final long processId;
-    private final OptimizingType optimizingType;
+    private OptimizingType optimizingType;
     private final TableRuntime tableRuntime;
     private final long planTime;
     private final long targetSnapshotId;
     private final long targetChangeSnapshotId;
     private final Map<OptimizingTaskId, TaskRuntime> taskMap = Maps.newHashMap();
     private final Queue<TaskRuntime> taskQueue = new LinkedList<>();
     private final Lock lock = new ReentrantLock();
-    private volatile Status status = OptimizingProcess.Status.RUNNING;
+    private volatile Status status = OptimizingProcess.Status.PLANNING;
     private volatile String failedReason;
     private long endTime = ArcticServiceConstants.INVALID_TIME;
     private Map<String, Long> fromSequence = Maps.newHashMap();
     private Map<String, Long> toSequence = Maps.newHashMap();
+    private final OptimizingPlanner planner;
```

Comment:
I think the main issue here is that the planner, as a field of the optimizing process, is not being persisted, which leads to the process having to be closed after restarting AMS. 
My suggestion is not to add this field `planner`, but instead to construct a new planner when calling `planTasks()` and pass it as a parameter, and the `targetSnapshotId` and `targetChangeSnapshotId` will be set when planning tasks.